### ISSUE TITLE
[JEI] Crafting station recipe transfer breaks when next to a container

### ISSUE THREAD

**Issue by TJT01** [SlimeKnights/TinkersConstruct#5250]
### Minecraft Version

1.19.2

### Forge Version

43.3.5

### Mantle Version

1.10.36

### Tinkers' Construct Version

3.8.3.39

### Describe your issue

Tinkers' JEI crafting station recipe transfer handler breaks when a container (chests, barrels...) is placed next to the crafting station.
The + button isn't visible, and this error gets spammed in the log:
```
[28Jul2024 21:56:28.412] [Render thread/ERROR] [mezz.jei.library.transfer.BasicRecipeTransferHandler/]: Recipe Transfer helper class slimeknights.tconstruct.plugin.jei.transfer.CraftingStationTransferInfo does not work for container class slimeknights.tconstruct.tables.menu.CraftingStationContainerMenu. The Recipe Transfer Helper references inventory slot index [10] but it does not allow modification.
```
It works fine if there isn't a connected container.

Logs: https://mclo.gs/h61c6lc

### Other mods

Just Enough Items 11.6.0.1024

### Tried reproducing with just Tinkers?

Yes

### Performance Enchancers

None of the above

### Searched for known issues?

Checked pinned issues, Searched open issues, Searched closed issues

**Comment by alphafalcon** [comment]
Also happens in All The Mods 9 - To the Sky
Relevant Versions:

Minecraft: 1.20.1
Forge: 47.2.20
JEI: 15.8.2.23

**Comment by KnightMiner** [comment]
> Also happens in All The Mods 9 - To the Sky Relevant Versions:
> 
> Minecraft: 1.20.1 Forge: 47.2.20 JEI: 15.8.2.23

No it does not, there is no tinkers construct for 1.20.1

**Comment by alphafalcon** [comment]
Apologies, you are correct.
There is a "Crafting Station" mod present in the pack that provides a crafting station, which behaves completely identical to TiCons and generates the same error message.

I suspect they share some code.

**Comment by alphafalcon** [comment]
https://www.curseforge.com/minecraft/mc-mods/crafting-station-jei-edition

This should be a drop-in replacement. The author of the original crafting station mod has declined to add compatibility for JEI.

**Comment by Fureniku** [comment]
> Apologies, you are correct. There is a "Crafting Station" mod present in the pack that provides a crafting station, which behaves completely identical to TiCons and generates the same error message.
> 
> I suspect they share some code.

Yeah, that's mine (built off an MIT fork of the crafting station). Looking into a fix now and happened to find this ticket so figured I'd let you know a fix is coming!

**Comment by KnightMiner** [comment]
Hmm, seems JEI made an API change, which breaks due to a weird forge decision. I can mostly fix this on Tinkers side, but might need a Mantle change to properly support said JEI API.

**Comment by KnightMiner** [comment]
Fixed in 3.8.4.50 for 1.19.2. Does not affect 1.18.2
